tests/torchtune/modules/test_kv_cache.py

```diff
@@ -151,3 +153,28 @@ def test_kv_cache_multiple_updates(self, kv_cache, k_vals_full, v_vals_full):
 
         assert torch.equal(expected_k_out, k_out)
         assert torch.equal(expected_v_out, v_out)
+
+    def test_kv_cache_no_recompiles(self, kv_cache, k_vals_full, v_vals_full):
```

Comment:
I'm not sure if we care about seeing this kind of thing here, open to thoughts @ebsmothers?

Reply:
i like this test!

Reply:
big big big big big big fan of this test

Reply:
Yeah this is awesome. Dumb q though: is this hardware-dependent in any way? Want to make sure this is reproducible. At least from my quick look through how it's used in PyTorch core it seems fine? If that's the case I would love to start incorporating this kind of test into our models too

Reply:
Yeah it should be device agnostic, I pulled it from core and compiled my tests against MPS/CPU/CUDA. I'd also be in favour of adding them, we have non-trivial dependencies on compile behaviour. Be happy to help here.